File: app/src/main/java/org/softeg/slartus/forpdaplus/mainnotifiers/NotifiersManager.java
package org.softeg.slartus.forpdaplus.mainnotifiers;/*
 * Created by slinkin on 10.07.2014.
 */

import android.app.Activity;
import android.app.AlertDialog;
import android.content.DialogInterface;

import com.afollestad.materialdialogs.MaterialDialog;

import org.softeg.slartus.forpdaplus.common.AppLog;

import java.util.ArrayList;

public class NotifiersManager implements DialogInterface.OnDismissListener {
    private Activity activity;
    private ArrayList<DialogInterface> dialogs = new ArrayList<>();

    public NotifiersManager(Activity activity) {

        this.activity = activity;
    }

    public void addNotifyDialog(MaterialDialog alertDialog) {
        try {
            alertDialog.setOnDismissListener(this);
            dialogs.add(alertDialog);
            if (dialogs.size() == 1)
                alertDialog.show();
        } catch (Throwable ex) {
            ex.printStackTrace();
        }
    }

    @Override
    public void onDismiss(DialogInterface dialogInterface) {
        try {
            if (dialogs.contains(dialogInterface))
                dialogs.remove(dialogInterface);
            if (dialogs.size() > 0)
                ((AlertDialog) dialogs.get(0)).show();
        } catch (Throwable ex) {
            AppLog.e(activity, ex);
        }
    }
}

File: app/src/main/java/org/softeg/slartus/forpdaplus/mainnotifiers/NotifiersManager.kt
package org.softeg.slartus.forpdaplus.mainnotifiers

/*
 * Created by slinkin on 10.07.2014.
 */

import android.content.DialogInterface

import com.afollestad.materialdialogs.MaterialDialog

import java.util.ArrayList

class NotifiersManager : DialogInterface.OnDismissListener {

    private val dialogs = ArrayList<DialogInterface>()
    private val lock = Any()
    fun addNotifyDialog(alertDialog: MaterialDialog) {
        try {
            alertDialog.setOnDismissListener(this)
            synchronized(lock) {
                dialogs.add(alertDialog)
                if (dialogs.size == 1)
                    alertDialog.show()
            }
        } catch (ex: Throwable) {
            ex.printStackTrace()
        }

    }

    override fun onDismiss(dialogInterface: DialogInterface) {
        try {
            synchronized(lock) {
                if (dialogs.contains(dialogInterface))
                    dialogs.remove(dialogInterface)
                if (dialogs.size > 0)
                    (dialogs[0] as MaterialDialog).show()
            }
        } catch (ex: Throwable) {
            ex.printStackTrace()
        }
    }
}
